### PR TITLE
ref: Migrate `configure_scope` in `src/sentry/utils`

### PR DESCRIPTION
**Pull request by szokeasaurusrex**
Replace deprecated `configure_scope` with new `Scope.get_isolation_scope()` API from Sentry SDK 2.0.

ref #73430

### CODE REVIEW

File: src/sentry/utils/sdk.py [hunk 1]
```diff
@@ -104,16 +104,16 @@ def is_current_event_safe():
...
     recursion if an attempt to send to Sentry was made.
     """
+    scope = Scope.get_isolation_scope()
 
-    with configure_scope() as scope:
-        # Scope was explicitly marked as unsafe
-        if scope._tags.get(UNSAFE_TAG):
-            return False
+    # Scope was explicitly marked as unsafe
+    if scope._tags.get(UNSAFE_TAG):
```
Comment (szokeasaurusrex) [inline src/sentry/utils/sdk.py:110]: I am unsure whether this code works as expected now that we have current scopes as well

Comment (szokeasaurusrex) [reply src/sentry/utils/sdk.py:110]: Although, it is out of scope for this PR to fix this here, I am wondering whether we should maybe make an issue for this?

File: src/sentry/utils/sdk.py [hunk 2]
```diff
@@ -490,44 +490,43 @@ def check_tag_for_scope_bleed(
...
-                # But if it currently is a list and `expected_value` is something *not* in that list,
-                # we're almost certainly dealing with scope bleed, so we should continue with our check.
-                current_org_list = scope._contexts.get("organization", {}).get(
-                    "multiple possible", []
-                )
-                if current_org_list and expected_value in current_org_list:
-                    return
+    scope = Scope.get_isolation_scope()
+
+    current_value = scope._tags.get(tag_key)
```
Comment (szokeasaurusrex) [inline src/sentry/utils/sdk.py:495]: Likewise unsure if this will work as expected with current scopes also existing

**Review by Swatinem — APPROVED** [review]
out of curiosity:
not quite sure if this is relevant for python (probably its not?), but for other languages the idea of using a callback/closure-based approach was that the given closure would never be executed in case the SDK has never been initialized.
I assume the new code will return a "dummy" scope instead which just ignores anything you set on it?

File: src/sentry/utils/sdk.py (hunk 1 above)
Comment (sentrivana) [reply src/sentry/utils/sdk.py:110]: I think as long as the tag is explicitly set on the isolation scope and then also retrieved from the isolation scope, this should work ok after your changes

**Review by sentrivana — APPROVED** [review]

**Comment by szokeasaurusrex** [comment]
@sentrivana, I had to make some changes to the tests, since they were depending on us using the old API. Could you please re-review this PR, focusing on the test changes, before I merge?

**Review by sentrivana — APPROVED** [review]
The new tests changes also lgtm

**Pull request merged**